packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -1003,9 +1003,17 @@ export class CrossChainMessenger {
    * @returns Current challenge period in seconds.
    */
```

Comment:
Slight preference to get the version of the oracle once and assign it to the class and then use that version because we will have more version dependent logic in the future

Reply:
Doing it in the constructor would be bad because some uses of the cross chain messenger initialize cross chain messenger ephemerally in a function instead of passing around a global object.   We would want to lazily cache it.

```typescript
public async getOracleVersion(): Promise<number> {
  if (!this.oracleVersion) this.oracleVersion = await this.contract.l1.L2OutputOracle.version() 
  return this.oracleVersion
}
```

I don't think we should do this though because we'd want to think through the implications of this for long running processes.   Very quickly we would want the cache to expire and the engineering effort / surface area for bugs becomes too big for such a tiny optimization

Reply:
Actually thinking about it more I wouldn't be opposed to doing ^

Reply:
I actually do think that's a good idea but even if we do do that I'd recommend separating that into it's own pr and keeping this pr focused